src/media/js/apps_buttons.js

```diff
@@ -47,18 +49,15 @@ define('apps_buttons',
 
         // If it's a paid app, ask the user to sign in first.
         if (product.receipt_required && !user.logged_in()) {
-            var payWindow;
-
-            if (!capabilities.navPay) {
-                console.log('Creating payment window ahead of time via user click');
-                payWindow = payments.utils.openWindow({url: ''});
-            }
-
             console.log('Install suspended; user needs to log in');
-            return login.login().done(function() {
-                if (payWindow) {
-                    payWindow.focus();
+            return login.login({retainPopup: true}).done(function() {
+
+                // Set flag so we know it's desktop
+                // and a popup was created.
+                if (!capabilities.navPay) {
+                    desktopLoginPopup = true;
                 }
+
```

Comment:
I know it's just a WiP but one thing you'll need here is something like:
````
login.login(...).done(...).fail(function() {
  login.closeWindow();
});
````

However, this makes me think it would be a better approach to pass a window object like ``login.login({loginWindow: newWindow})`` similar to how you would pass one to fxpay. This might also allow you to get rid of the ``z.fxa_popup`` global.